cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,396 @@
+						},
+					},
+					SyncSets: []v1alpha1.SyncStatus{
+						{
+							Name:           "syncset1",
+							Result:         "Failure",
+							FailureMessage: "Failed to sync syncset1",
+						},
+					},
+					SelectorSyncSets: []v1alpha1.SyncStatus{
+						{
+							Name:           "selectorsyncset1",
+							Result:         "Failure",
+							FailureMessage: "Failed to sync selectorsyncset1",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+// Central function to mock the List method and set expectations
+func setupMockClient(mockClient *mockk8s.MockClient, returnErr error, isEmpty bool) {
+	callTimes := 1
+	if returnErr == nil {
+		callTimes = 2 // We expect two calls only if there's no error
+	}
+	mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+			switch v := list.(type) {
+			case *hivev1.ClusterDeploymentList:
+				*v = cdList
+			case *v1alpha1.ClusterSyncList:
+				if isEmpty {
+					*v = v1alpha1.ClusterSyncList{} // Return empty ClusterSyncList for the empty case
+				} else {
+					*v = csList // Assuming csList is populated normally
+				}
+			}
+			return returnErr
+		}).Times(callTimes) // Expect List to be called n times based on the error condition
+}
+
+func TestPrintFailingCluster(t *testing.T) {
+	tests := []struct {
+		name        string
+		cdList      *hivev1.ClusterDeploymentList
+		csList      *v1alpha1.ClusterSyncList
+		expectError bool
+	}{
+		{
+			name: "Successful_Execution",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			expectError: false,
+		},
+		{
+			name: "Missing_ClusterDeployment",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{},
+			},
+			expectError: true,
+		},
+		{
+			name: "SyncSet_Failure",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+					Status: v1alpha1.ClusterSyncStatus{
+						SyncSets: []v1alpha1.SyncStatus{{
+							Name:           "sync-failure",
+							Result:         "Failure",
+							FailureMessage: "Some error occurred",
+						}},
+					},
+				}},
+			},
+			expectError: false,
```

Comment:
The verbiage on this test is confusing, b/c it's a failure state but `expectError` is set to `false`.

Having looked at the package code, it's unfortunate that the function being tested isn't returning the `FailureMessage` so that we can check it.

Because of that, perhaps we shouldn't have a `Result: "Failure"` because it's confusing when reading this (and not looking at the code) why we wouldn't be expecting an error.

Reply:
I'm keeping this test to ensure the `printFailingCluster ` method correctly prints failure messages. While it doesn’t return errors, the test verifies that the appropriate messages are printed for failure scenarios. Added a comment that clarifies that failures are handled by printing, not returning errors.

Reply:
@drakshakhan Please don't resolve these, it's difficult to find them again :)

You're correct, but this test isn't able to verify the printed messages.  Because of that, it's confusing that it's purportedly testing an error but `expectErr` is `false`.

Reply:
I understand your concern. I’ve added a comment clarifying that the function prints errors instead of returning them. Would that be sufficient, or do you think it would be better to remove this test case altogether?